data/mods/Magiclysm/Spells/animist.json

```diff
@@ -69,148 +67,315 @@
     "final_energy_cost": 10000
   },
   {
-    "id": "summon_zombie",
+    "id": "summon_undead",
     "type": "SPELL",
-    "name": "Summon Zombie",
-    "description": "An ethereal-looking zombie rises from the depths of the earth to fight for you.  You may be able to summon more with a higher level in this spell.",
-    "valid_targets": [ "ground" ],
+    "name": "Summon Undead",
+    "description": "An ethereal-looking undead rises from the depths of the earth to fight for you.  You may be able to summon the stronger one with a higher level in this spell.",
+    "valid_targets": [ "self" ],
     "min_damage": 1,
-    "max_damage": 3,
-    "//": "level 20 will max out your number of summons",
-    "damage_increment": 0.1,
-    "max_level": 30,
-    "difficulty": 2,
+    "max_damage": 1,
+    "max_level": 35,
+    "difficulty": 3,
     "spell_class": "ANIMIST",
-    "effect": "summon",
-    "effect_str": "mon_zombie",
+    "effect": "effect_on_condition",
+    "effect_str": "EOC_SUMMON_ZOMBIE",
+    "components": "spell_components_rune_animist",
     "shape": "blast",
-    "energy_source": "HP",
-    "base_energy_cost": 6,
-    "//2": "half a minute",
+    "energy_source": "MANA",
+    "base_energy_cost": 400,
     "base_casting_time": 3000,
-    "min_range": 3,
-    "max_range": 3,
-    "min_aoe": 2,
-    "max_aoe": 2,
     "flags": [ "SOMATIC", "VERBAL", "CONCENTRATE" ],
     "min_duration": 36000,
     "max_duration": 1080000,
-    "duration_increment": 36000,
-    "learn_spells": { "summon_tough_zombie": 20 }
+    "duration_increment": 36000
   },
   {
-    "id": "summon_tough_zombie",
+    "type": "effect_on_condition",
+    "id": "EOC_SUMMON_ZOMBIE",
+    "effect": {
+      "switch": { "u_val": "spell_level", "spell": "summon_undead" },
```

Comment:
Would this work? I don't know how the conversion to code is for this, but if this was written directly in code it would only summon on the levels specified in the cases. A zombie would be summoned for level 0, but not for level 1. Might even force a crash.